src/main/java/jarvis/model/LessonNotes.java

```diff
@@ -46,6 +47,54 @@ public void addNote(Student student, String notes) {
         studentNotes.get(student).append("\n");
     }
 
+    /**
+     * Deletes note at given index from overall lesson notes.
+     *
+     * @param index Index of note according to order of notes in overall notes.
+     * @return String of the deleted note.
+     */
+    public String deleteNote(int index) {
+        int deleteStartIndex = 0;
+        int deleteEndIndex = 0;
+        for (int i = 0; i <= index; i++) {
+            deleteStartIndex = deleteEndIndex;
+            deleteEndIndex = overallNotes.indexOf("\n", deleteStartIndex) + 1;
```

Comment:
This part seems very prone to bugs. If we are planning to index the notes, I think a better design would be to represent the notes as an `ArrayList` where each element in the list is a line. We can then index the list properly.

Reply:
Was hoping to reduce conflicts since other changes might also be made based on the implementation of LessonNotes. Will update this next time.